examples/expressjs-ethr/src/index.ts

```diff
@@ -0,0 +1,224 @@
+
+    // Counting views to show that session is working
+    req.session.views = req.session.views ? req.session.views + 1 : 1
+
+    const { did, views } = req.session
+    const name = await dataStore.shortId(did)
+    res.render('home', { did, views, name })
+  })
+
+  app.get('/history', requireLogin, async (req, res) => {
+    if (!req.session) throw Error('Session not configured')
+
+    // Counting views to show that session is working
+    req.session.views = req.session.views ? req.session.views + 1 : 1
+
+    const { did, views } = req.session
+    const name = await dataStore.shortId(did)
+    const messages = await dataStore.findMessages({ sender: did })
+    console.log(messages)
+    res.render('history', { did, views, name, messages })
+  })
+
+  app.get('/login', async (req, res) => {
+    if (!req.session) throw Error('Session not configured')
+
+    // Counting views to show that session is working
+    req.session.views = req.session.views ? req.session.views + 1 : 1
+
+    // Sign Selective Disclosure Request
+    const jwt = await core.handleAction({
+      type: SD.ActionTypes.signSdr,
+      did: identity.did,
+      data: {
+        tag: req.sessionID,
+        claims: [
+          {
+            reason: 'We need this information',
+            essential: true,
+            claimType: 'name',
+          },
+        ],
+      },
+    } as SD.ActionSignSdr)
+
+    const url = encodeURI(process.env.HOST + '/?c_i=') + jwt
+
+    res.render('login', { views: req.session.views, url })
+  })
+
+  app.get('/credential', requireLogin, async (req, res) => {
+    if (!req.session) throw Error('Session not configured')
+
+    // Counting views to show that session is working
+    req.session.views = req.session.views ? req.session.views + 1 : 1
+
+    const { did, views } = req.session
+    const name = await dataStore.shortId(did)
+
+    // Sign verifiable credential
+    const nameJwt = await core.handleAction({
+      type: W3C.ActionTypes.signVc,
+      did: identity.did,
+      data: {
+        sub: did,
+        vc: {
+          '@context': ['https://www.w3.org/2018/credentials/v1'],
+          type: ['VerifiableCredential'],
+          credentialSubject: {
+            name,
+          },
+        },
+      },
+    } as W3C.ActionSignW3cVc)
+
+    const kwcJwt = await core.handleAction({
+      type: W3C.ActionTypes.signVc,
+      did: identity.did,
+      data: {
+        sub: did,
+        vc: {
+          '@context': ['https://www.w3.org/2018/credentials/v1'],
+          type: ['VerifiableCredential'],
+          credentialSubject: {
+            kycId: '123XZY',
+          },
+        },
+      },
+    } as W3C.ActionSignW3cVc)
+
+    const vpJwt = await core.handleAction({
+      type: W3C.ActionTypes.signVp,
+      did: identity.did,
+      data: {
+        sub: did,
+        vp: {
+          '@context': ['https://www.w3.org/2018/credentials/v1'],
+          type: ['VerifiableCredential'],
```

Comment:
Should this not be a presentation?

Reply:
Good catch.

cc @simonas-notcat

Reply:
@mirceanis we need to fix validation logic in [`did-jwt-vc`](https://github.com/decentralized-identity/did-jwt-vc/blob/develop/src/__tests__/index.ts#L38).

Reply:
@mirceanis I created this PR to fix VP issue: https://github.com/decentralized-identity/did-jwt-vc/pull/20